### PR TITLE
Feature/mde usability

### PR DESCRIPTION
**Pull request by Dmatryus**

### CODE REVIEW

File: hypex/ab_test/aa_tester.py [hunk 1]
```diff
@@ -889,6 +762,176 @@ def cat_feature_uniform_analysis(
...
+
+        return p * m * s
+
+    def calc_sample_size(
+        self,
+        test_group: pd.Series = None,
+        control_group: pd.Series = None,
+        data: pd.DataFrame = None,
+        target_field: str = None,
+        group_col: str = "group",
```
Comment (VasilievaKA) [inline hypex/ab_test/aa_tester.py:839]: No usage of this variable in functions' body

File: hypex/ab_test/aa_tester.py [hunk 2]
```diff
@@ -906,7 +949,28 @@ def experiment_result_transform(self, experiment: pd.Series):
...
                 if i.startswith(f"{tf} "):
                     targets_dict[tf][i[len(tf) + 1 :]] = experiment[i]
-        return pd.DataFrame(targets_dict).T, experiment.iloc[-9:]
+
+        return {
+            "best_experiment_stat": pd.DataFrame(targets_dict).T,
+            "best_split_stat": experiment.iloc[-9:],
+        }
+
+    @staticmethod
```
Comment (VasilievaKA) [inline hypex/ab_test/aa_tester.py:958]: line 984 
now it must be ssp = self.split_splited_data(splited_data)

File: examples/tutorials/Tutorial_2_AAtesting.ipynb [hunk 3]
```diff
@@ -30,14 +30,15 @@
...
     "import numpy as np\n",
+    "import pandas as pd\n",
+    "\n",
     "from hypex import AATest\n",
     "from hypex.utils.tutorial_data_creation import create_test_data\n",
     "\n",
     "pd.options.display.float_format = '{:,.2f}'.format\n",
     "\n",
-    "np.random.seed(52) # needed to create example data"
+    "np.random.seed(52)  # needed to create example data"
```
Comment (tikhomirovd) [inline examples/tutorials/Tutorial_2_AAtesting.ipynb:41]: please use random seed 42, cause to Douglas Adams

**Comment by tikhomirovd** [comment]
### Description

This PR introduces significant enhancements to the AATest class within the HypEx library. The main focus is on improving the usability, adding new functionalities, and ensuring that the code adheres to modern Python standards. These changes aim to make AATest more intuitive for users, particularly in handling data inputs and calculating key statistical measures.

### Changes Made

- **Refactoring of `calc_mde`, `calc_sample_size`, and `calc_power` Methods:** These functions have been transformed into methods of the AATest class to streamline the user experience and reduce the need for external imports.
- **Alternative Data Input for `calc_mde` and `calc_sample_size`:** Added functionality to input data directly as a DataFrame along with specifying the target column. This enhancement caters to the common workflow of users post-group splitting.
- **Integration of an Automated Testing Suite:** To ensure the reliability of `calc_mde` and `calc_sample_size` functions, an automated test suite has been implemented.
- **Enhanced Tutorial Documentation:** The AA testing tutorial now includes a section demonstrating the application of `calc_mde` and `calc_sample_size` functions, providing users with practical examples of their usage.

### Related Issues

- Refinement of user input handling.
- Enhancement of the documentation and examples.

### Additional Notes

The refactoring efforts and the introduction of new features are expected to make the library more accessible to users, encouraging broader adoption and facilitating more robust statistical analysis within the banking sector.

### Testing and Validation

The new features and changes have been extensively tested with synthetic datasets to ensure accuracy and reliability. The automated test suite has been expanded to cover the new functionalities introduced in this PR.

### Performance Considerations

The changes introduced have been optimized for performance, ensuring that the enhancements do not negatively impact the execution time of analyses conducted using the AATest class.

### Breaking Changes

This PR does not introduce any breaking changes. All modifications are backward compatible with existing implementations of the HypEx library.

### Dependencies

No new dependencies have been introduced in this PR. The existing dependencies remain sufficient to support the new features and functionalities.

### Merge Request Checklist

- [x] Code follows project coding guidelines.
- [x] Documentation reflects the changes made.
- [x] Unit tests cover new or changed functionality.
- [x] Performance and breaking changes have been considered.

File: hypex/ab_test/aa_tester.py (hunk 1 above)
Comment (Dmatryus) [reply hypex/ab_test/aa_tester.py:839]: Yes. This is bug. I will fix it.

File: hypex/ab_test/aa_tester.py (hunk 2 above)
Comment (Dmatryus) [reply hypex/ab_test/aa_tester.py:958]: It's strange that the test passed. I'll fix it.

**Review by tikhomirovd — APPROVED** [review]
Great one

**Pull request merged**